diesel_derives/src/model.rs

```diff
@@ -55,6 +62,29 @@ impl Model {
         let model_name = self.name.as_ref().to_uppercase();
         format!("_IMPL_{}_FOR_{}", trait_name, model_name).into()
     }
+
+    pub fn destruct_pattern(&self) -> ::quote::Tokens {
```

Comment:
We don't need this method. `let #struct_name { #(#field_name: #column_name,)* }` works in all cases. Alternatively, just skip destructuring entirely (which was mostly done to make the `macro_rules!` form easier), and use `self.#field_name`